webapp/channels/src/actions/file_actions.ts

```diff
@@ -87,6 +87,20 @@ export function uploadFile({file, name, type, rootId, channelId, clientId, onPro
                     ]));
 
                     onSuccess(response, channelId, rootId);
+                } else if (xhr.readyState === 4 && xhr.status === 400) {
+                    const errorResponse = JSON.parse(xhr.response);
+                    const errorMessage =
+                        (errorResponse?.id && errorResponse?.message) ? localizeMessage(errorResponse.id, errorResponse.message) :
+                            localizeMessage('file_upload.generic_error', 'There was a problem uploading your files.');
+
+                    dispatch({
+                        type: FileTypes.UPLOAD_FILES_FAILURE,
+                        clientIds: [clientId],
+                        channelId,
+                        rootId,
+                    });
+
+                    onError(errorMessage, clientId, channelId, rootId);
```

Comment:
Is `onError` always present? We have an if statement checking for it below so maybe not

Reply:
I do believe it will always be present. It's defined [here](https://github.com/MattSilvaa/mattermost-server/blob/master/webapp/channels/src/components/file_upload/file_upload.tsx#L285) and the `uploadFile` interface indicates that both `onSuccess` and `onError` must always be defined. I am going to remove the if statement for both as it's unnecessary.

Reply:
I will modify it to be `onError?.(errorMessage, clientId, channelId, rootId)` to be defensive in the scenario where it isn't defined